// @paypal/paypal-server-sdk/dist/esm/models/applePayCardResponse.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { authenticationResponseSchema, } from './authenticationResponse.js';
import { binDetailsSchema } from './binDetails.js';
import { cardAttributesResponseSchema, } from './cardAttributesResponse.js';
import { cardBrandSchema } from './cardBrand.js';
import { cardFromRequestSchema } from './cardFromRequest.js';
import { cardStoredCredentialSchema, } from './cardStoredCredential.js';
import { cardTypeSchema } from './cardType.js';
export const applePayCardResponseSchema = object({
    name: ['name', optional(string())],
    lastDigits: ['last_digits', optional(string())],
    brand: ['brand', optional(cardBrandSchema)],
    availableNetworks: ['available_networks', optional(array(cardBrandSchema))],
    type: ['type', optional(cardTypeSchema)],
    authenticationResult: [
        'authentication_result',
        optional(lazy(() => authenticationResponseSchema)),
    ],
    attributes: [
        'attributes',
        optional(lazy(() => cardAttributesResponseSchema)),
    ],
    fromRequest: ['from_request', optional(lazy(() => cardFromRequestSchema))],
    expiry: ['expiry', optional(string())],
    binDetails: ['bin_details', optional(lazy(() => binDetailsSchema))],
    storedCredential: [
        'stored_credential',
        optional(lazy(() => cardStoredCredentialSchema)),
    ],
    billingAddress: ['billing_address', optional(lazy(() => addressSchema))],
    countryCode: ['country_code', optional(string())],
});
