// @paypal/paypal-server-sdk/dist/esm/models/applePayPaymentObject.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { applePayAttributesResponseSchema, } from './applePayAttributesResponse.js';
import { applePayCardResponseSchema, } from './applePayCardResponse.js';
import { cardStoredCredentialSchema, } from './cardStoredCredential.js';
import { phoneNumberSchema } from './phoneNumber.js';
export const applePayPaymentObjectSchema = object({
    id: ['id', optional(string())],
    token: ['token', optional(string())],
    name: ['name', optional(string())],
    emailAddress: ['email_address', optional(string())],
    phoneNumber: ['phone_number', optional(lazy(() => phoneNumberSchema))],
    card: ['card', optional(lazy(() => applePayCardResponseSchema))],
    attributes: [
        'attributes',
        optional(lazy(() => applePayAttributesResponseSchema)),
    ],
    storedCredential: [
        'stored_credential',
        optional(lazy(() => cardStoredCredentialSchema)),
    ],
});
