// @paypal/paypal-server-sdk/dist/esm/models/applePayRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { applePayAttributesSchema, } from './applePayAttributes.js';
import { applePayDecryptedTokenDataSchema, } from './applePayDecryptedTokenData.js';
import { applePayExperienceContextSchema, } from './applePayExperienceContext.js';
import { cardStoredCredentialSchema, } from './cardStoredCredential.js';
import { phoneNumberSchema } from './phoneNumber.js';
export const applePayRequestSchema = object({
    id: ['id', optional(string())],
    name: ['name', optional(string())],
    emailAddress: ['email_address', optional(string())],
    phoneNumber: ['phone_number', optional(lazy(() => phoneNumberSchema))],
    decryptedToken: [
        'decrypted_token',
        optional(lazy(() => applePayDecryptedTokenDataSchema)),
    ],
    storedCredential: [
        'stored_credential',
        optional(lazy(() => cardStoredCredentialSchema)),
    ],
    vaultId: ['vault_id', optional(string())],
    attributes: ['attributes', optional(lazy(() => applePayAttributesSchema))],
    experienceContext: [
        'experience_context',
        optional(lazy(() => applePayExperienceContextSchema)),
    ],
});
