// @paypal/paypal-server-sdk/dist/esm/models/billingPlan.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, object, optional, string, } from '../schema.js';
import { linkDescriptionSchema } from './linkDescription.js';
import { merchantPreferencesSchema, } from './merchantPreferences.js';
import { paymentPreferencesSchema, } from './paymentPreferences.js';
import { subscriptionBillingCycleSchema, } from './subscriptionBillingCycle.js';
import { subscriptionPlanStatusSchema, } from './subscriptionPlanStatus.js';
import { taxesSchema } from './taxes.js';
export const billingPlanSchema = object({
    id: ['id', optional(string())],
    productId: ['product_id', optional(string())],
    name: ['name', optional(string())],
    status: ['status', optional(subscriptionPlanStatusSchema)],
    description: ['description', optional(string())],
    billingCycles: [
        'billing_cycles',
        optional(array(lazy(() => subscriptionBillingCycleSchema))),
    ],
    paymentPreferences: [
        'payment_preferences',
        optional(lazy(() => paymentPreferencesSchema)),
    ],
    merchantPreferences: [
        'merchant_preferences',
        optional(lazy(() => merchantPreferencesSchema)),
    ],
    taxes: ['taxes', optional(lazy(() => taxesSchema))],
    quantitySupported: ['quantity_supported', optional(boolean())],
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
});
