// @paypal/paypal-server-sdk/dist/esm/models/cardRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { cardAttributesSchema } from './cardAttributes.js';
import { cardExperienceContextSchema, } from './cardExperienceContext.js';
import { cardStoredCredentialSchema, } from './cardStoredCredential.js';
import { networkTokenSchema } from './networkToken.js';
export const cardRequestSchema = object({
    name: ['name', optional(string())],
    number: ['number', optional(string())],
    expiry: ['expiry', optional(string())],
    securityCode: ['security_code', optional(string())],
    billingAddress: ['billing_address', optional(lazy(() => addressSchema))],
    attributes: ['attributes', optional(lazy(() => cardAttributesSchema))],
    vaultId: ['vault_id', optional(string())],
    singleUseToken: ['single_use_token', optional(string())],
    storedCredential: [
        'stored_credential',
        optional(lazy(() => cardStoredCredentialSchema)),
    ],
    networkToken: ['network_token', optional(lazy(() => networkTokenSchema))],
    experienceContext: [
        'experience_context',
        optional(lazy(() => cardExperienceContextSchema)),
    ],
});
