// @paypal/paypal-server-sdk/dist/esm/models/cardBrand.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { stringEnum } from '../schema.js';
/**
 * Enum for CardBrand
 */
export var CardBrand;
(function (CardBrand) {
    CardBrand["Visa"] = "VISA";
    CardBrand["Mastercard"] = "MASTERCARD";
    CardBrand["Discover"] = "DISCOVER";
    CardBrand["Amex"] = "AMEX";
    CardBrand["Solo"] = "SOLO";
    CardBrand["Jcb"] = "JCB";
    CardBrand["Star"] = "STAR";
    CardBrand["Delta"] = "DELTA";
    CardBrand["Switch"] = "SWITCH";
    CardBrand["Maestro"] = "MAESTRO";
    CardBrand["CbNationale"] = "CB_NATIONALE";
    CardBrand["Configoga"] = "CONFIGOGA";
    CardBrand["Confidis"] = "CONFIDIS";
    CardBrand["Electron"] = "ELECTRON";
    CardBrand["Cetelem"] = "CETELEM";
    CardBrand["ChinaUnionPay"] = "CHINA_UNION_PAY";
    CardBrand["Diners"] = "DINERS";
    CardBrand["Elo"] = "ELO";
    CardBrand["Hiper"] = "HIPER";
    CardBrand["Hipercard"] = "HIPERCARD";
    CardBrand["Rupay"] = "RUPAY";
    CardBrand["Ge"] = "GE";
    CardBrand["Synchrony"] = "SYNCHRONY";
    CardBrand["Eftpos"] = "EFTPOS";
    CardBrand["CarteBancaire"] = "CARTE_BANCAIRE";
    CardBrand["StarAccess"] = "STAR_ACCESS";
    CardBrand["Pulse"] = "PULSE";
    CardBrand["Nyce"] = "NYCE";
    CardBrand["Accel"] = "ACCEL";
    CardBrand["Unknown"] = "UNKNOWN";
})(CardBrand || (CardBrand = {}));
/**
 * Schema for CardBrand
 */
export const cardBrandSchema = stringEnum(CardBrand, true);
