// @paypal/paypal-server-sdk/dist/esm/models/itemRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { itemCategorySchema } from './itemCategory.js';
import { moneySchema } from './money.js';
import { orderBillingPlanSchema, } from './orderBillingPlan.js';
import { universalProductCodeSchema, } from './universalProductCode.js';
export const itemRequestSchema = object({
    name: ['name', string()],
    unitAmount: ['unit_amount', lazy(() => moneySchema)],
    tax: ['tax', optional(lazy(() => moneySchema))],
    quantity: ['quantity', string()],
    description: ['description', optional(string())],
    sku: ['sku', optional(string())],
    url: ['url', optional(string())],
    category: ['category', optional(itemCategorySchema)],
    imageUrl: ['image_url', optional(string())],
    upc: ['upc', optional(lazy(() => universalProductCodeSchema))],
    billingPlan: ['billing_plan', optional(lazy(() => orderBillingPlanSchema))],
});
