// @paypal/paypal-server-sdk/dist/esm/models/itemDetails.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { checkoutOptionSchema } from './checkoutOption.js';
import { moneySchema } from './money.js';
import { taxAmountSchema } from './taxAmount.js';
export const itemDetailsSchema = object({
    itemCode: ['item_code', optional(string())],
    itemName: ['item_name', optional(string())],
    itemDescription: ['item_description', optional(string())],
    itemOptions: ['item_options', optional(string())],
    itemQuantity: ['item_quantity', optional(string())],
    itemUnitPrice: ['item_unit_price', optional(lazy(() => moneySchema))],
    itemAmount: ['item_amount', optional(lazy(() => moneySchema))],
    discountAmount: ['discount_amount', optional(lazy(() => moneySchema))],
    adjustmentAmount: ['adjustment_amount', optional(lazy(() => moneySchema))],
    giftWrapAmount: ['gift_wrap_amount', optional(lazy(() => moneySchema))],
    taxPercentage: ['tax_percentage', optional(string())],
    taxAmounts: ['tax_amounts', optional(array(lazy(() => taxAmountSchema)))],
    basicShippingAmount: [
        'basic_shipping_amount',
        optional(lazy(() => moneySchema)),
    ],
    extraShippingAmount: [
        'extra_shipping_amount',
        optional(lazy(() => moneySchema)),
    ],
    handlingAmount: ['handling_amount', optional(lazy(() => moneySchema))],
    insuranceAmount: ['insurance_amount', optional(lazy(() => moneySchema))],
    totalItemAmount: ['total_item_amount', optional(lazy(() => moneySchema))],
    invoiceNumber: ['invoice_number', optional(string())],
    checkoutOptions: [
        'checkout_options',
        optional(array(lazy(() => checkoutOptionSchema))),
    ],
});
