// @paypal/paypal-server-sdk/dist/esm/models/orderAuthorizeResponse.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { checkoutPaymentIntentSchema, } from './checkoutPaymentIntent.js';
import { linkDescriptionSchema } from './linkDescription.js';
import { orderAuthorizeResponsePaymentSourceSchema, } from './orderAuthorizeResponsePaymentSource.js';
import { orderStatusSchema } from './orderStatus.js';
import { payerSchema } from './payer.js';
import { purchaseUnitSchema } from './purchaseUnit.js';
export const orderAuthorizeResponseSchema = object({
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    id: ['id', optional(string())],
    paymentSource: [
        'payment_source',
        optional(lazy(() => orderAuthorizeResponsePaymentSourceSchema)),
    ],
    intent: ['intent', optional(checkoutPaymentIntentSchema)],
    payer: ['payer', optional(lazy(() => payerSchema))],
    purchaseUnits: [
        'purchase_units',
        optional(array(lazy(() => purchaseUnitSchema))),
    ],
    status: ['status', optional(orderStatusSchema)],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
});
