// @paypal/paypal-server-sdk/dist/esm/models/paymentAuthorization.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { authorizationStatusSchema, } from './authorizationStatus.js';
import { authorizationStatusDetailsSchema, } from './authorizationStatusDetails.js';
import { linkDescriptionSchema } from './linkDescription.js';
import { moneySchema } from './money.js';
import { networkTransactionSchema, } from './networkTransaction.js';
import { payeeBaseSchema } from './payeeBase.js';
import { paymentSupplementaryDataSchema, } from './paymentSupplementaryData.js';
import { sellerProtectionSchema, } from './sellerProtection.js';
export const paymentAuthorizationSchema = object({
    status: ['status', optional(authorizationStatusSchema)],
    statusDetails: [
        'status_details',
        optional(lazy(() => authorizationStatusDetailsSchema)),
    ],
    id: ['id', optional(string())],
    amount: ['amount', optional(lazy(() => moneySchema))],
    invoiceId: ['invoice_id', optional(string())],
    customId: ['custom_id', optional(string())],
    networkTransactionReference: [
        'network_transaction_reference',
        optional(lazy(() => networkTransactionSchema)),
    ],
    sellerProtection: [
        'seller_protection',
        optional(lazy(() => sellerProtectionSchema)),
    ],
    expirationTime: ['expiration_time', optional(string())],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    supplementaryData: [
        'supplementary_data',
        optional(lazy(() => paymentSupplementaryDataSchema)),
    ],
    payee: ['payee', optional(lazy(() => payeeBaseSchema))],
});
