// @paypal/paypal-server-sdk/dist/esm/models/planRequest.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, object, optional, string, } from '../schema.js';
import { merchantPreferencesSchema, } from './merchantPreferences.js';
import { paymentPreferencesSchema, } from './paymentPreferences.js';
import { planRequestStatusSchema, } from './planRequestStatus.js';
import { subscriptionBillingCycleSchema, } from './subscriptionBillingCycle.js';
import { taxesSchema } from './taxes.js';
export const planRequestSchema = object({
    productId: ['product_id', string()],
    name: ['name', string()],
    status: ['status', optional(planRequestStatusSchema)],
    description: ['description', optional(string())],
    billingCycles: [
        'billing_cycles',
        array(lazy(() => subscriptionBillingCycleSchema)),
    ],
    paymentPreferences: [
        'payment_preferences',
        lazy(() => paymentPreferencesSchema),
    ],
    merchantPreferences: [
        'merchant_preferences',
        optional(lazy(() => merchantPreferencesSchema)),
    ],
    taxes: ['taxes', optional(lazy(() => taxesSchema))],
    quantitySupported: ['quantity_supported', optional(boolean())],
});
