// @paypal/paypal-server-sdk/dist/esm/models/purchaseUnit.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string, unknown, } from '../schema.js';
import { amountWithBreakdownSchema, } from './amountWithBreakdown.js';
import { itemSchema } from './item.js';
import { payeeBaseSchema } from './payeeBase.js';
import { paymentCollectionSchema, } from './paymentCollection.js';
import { paymentInstructionSchema, } from './paymentInstruction.js';
import { shippingWithTrackingDetailsSchema, } from './shippingWithTrackingDetails.js';
import { supplementaryDataSchema, } from './supplementaryData.js';
export const purchaseUnitSchema = object({
    referenceId: ['reference_id', optional(string())],
    amount: ['amount', optional(lazy(() => amountWithBreakdownSchema))],
    payee: ['payee', optional(lazy(() => payeeBaseSchema))],
    paymentInstruction: [
        'payment_instruction',
        optional(lazy(() => paymentInstructionSchema)),
    ],
    description: ['description', optional(string())],
    customId: ['custom_id', optional(string())],
    invoiceId: ['invoice_id', optional(string())],
    id: ['id', optional(string())],
    softDescriptor: ['soft_descriptor', optional(string())],
    items: ['items', optional(array(lazy(() => itemSchema)))],
    shipping: [
        'shipping',
        optional(lazy(() => shippingWithTrackingDetailsSchema)),
    ],
    supplementaryData: [
        'supplementary_data',
        optional(lazy(() => supplementaryDataSchema)),
    ],
    payments: ['payments', optional(lazy(() => paymentCollectionSchema))],
    mostRecentErrors: ['most_recent_errors', optional(array(unknown()))],
});
