// @paypal/paypal-server-sdk/dist/esm/models/setupTokenRequestCard.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { cardBrandSchema } from './cardBrand.js';
import { vaultCardExperienceContextSchema, } from './vaultCardExperienceContext.js';
import { vaultCardVerificationMethodSchema, } from './vaultCardVerificationMethod.js';
export const setupTokenRequestCardSchema = object({
    name: ['name', optional(string())],
    number: ['number', optional(string())],
    expiry: ['expiry', optional(string())],
    securityCode: ['security_code', optional(string())],
    brand: ['brand', optional(cardBrandSchema)],
    billingAddress: ['billing_address', optional(lazy(() => addressSchema))],
    verificationMethod: [
        'verification_method',
        optional(vaultCardVerificationMethodSchema),
    ],
    experienceContext: [
        'experience_context',
        optional(lazy(() => vaultCardExperienceContextSchema)),
    ],
});
