// @paypal/paypal-server-sdk/dist/esm/models/setupTokenResponseCard.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { binDetailsSchema } from './binDetails.js';
import { cardAuthenticationResponseSchema, } from './cardAuthenticationResponse.js';
import { cardBrandSchema } from './cardBrand.js';
import { cardResponseAddressSchema, } from './cardResponseAddress.js';
import { cardTypeSchema } from './cardType.js';
import { cardVerificationDetailsSchema, } from './cardVerificationDetails.js';
import { cardVerificationStatusSchema, } from './cardVerificationStatus.js';
import { networkTransactionReferenceEntitySchema, } from './networkTransactionReferenceEntity.js';
export const setupTokenResponseCardSchema = object({
    name: ['name', optional(string())],
    lastDigits: ['last_digits', optional(string())],
    brand: ['brand', optional(cardBrandSchema)],
    expiry: ['expiry', optional(string())],
    billingAddress: [
        'billing_address',
        optional(lazy(() => cardResponseAddressSchema)),
    ],
    verificationStatus: [
        'verification_status',
        optional(cardVerificationStatusSchema),
    ],
    verification: [
        'verification',
        optional(lazy(() => cardVerificationDetailsSchema)),
    ],
    networkTransactionReference: [
        'network_transaction_reference',
        optional(lazy(() => networkTransactionReferenceEntitySchema)),
    ],
    authenticationResult: [
        'authentication_result',
        optional(lazy(() => cardAuthenticationResponseSchema)),
    ],
    binDetails: ['bin_details', optional(lazy(() => binDetailsSchema))],
    type: ['type', optional(cardTypeSchema)],
});
