// @paypal/paypal-server-sdk/dist/esm/models/setupTokenRequestPaymentSource.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional } from '../schema.js';
import { bankRequestSchema } from './bankRequest.js';
import { setupTokenRequestCardSchema, } from './setupTokenRequestCard.js';
import { vaultApplePayRequestSchema, } from './vaultApplePayRequest.js';
import { vaultPaypalWalletRequestSchema, } from './vaultPaypalWalletRequest.js';
import { vaultTokenRequestSchema, } from './vaultTokenRequest.js';
import { vaultVenmoRequestSchema, } from './vaultVenmoRequest.js';
export const setupTokenRequestPaymentSourceSchema = object({
    card: ['card', optional(lazy(() => setupTokenRequestCardSchema))],
    paypal: ['paypal', optional(lazy(() => vaultPaypalWalletRequestSchema))],
    venmo: ['venmo', optional(lazy(() => vaultVenmoRequestSchema))],
    applePay: ['apple_pay', optional(lazy(() => vaultApplePayRequestSchema))],
    token: ['token', optional(lazy(() => vaultTokenRequestSchema))],
    bank: ['bank', optional(lazy(() => bankRequestSchema))],
});
