// @paypal/paypal-server-sdk/dist/esm/models/shippingWithTrackingDetails.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { fulfillmentTypeSchema } from './fulfillmentType.js';
import { orderTrackerResponseSchema, } from './orderTrackerResponse.js';
import { phoneNumberWithOptionalCountryCodeSchema, } from './phoneNumberWithOptionalCountryCode.js';
import { shippingNameSchema } from './shippingName.js';
import { shippingOptionSchema } from './shippingOption.js';
export const shippingWithTrackingDetailsSchema = object({
    trackers: [
        'trackers',
        optional(array(lazy(() => orderTrackerResponseSchema))),
    ],
    name: ['name', optional(lazy(() => shippingNameSchema))],
    emailAddress: ['email_address', optional(string())],
    phoneNumber: [
        'phone_number',
        optional(lazy(() => phoneNumberWithOptionalCountryCodeSchema)),
    ],
    type: ['type', optional(fulfillmentTypeSchema)],
    options: ['options', optional(array(lazy(() => shippingOptionSchema)))],
    address: ['address', optional(lazy(() => addressSchema))],
});
