// @paypal/paypal-server-sdk/dist/esm/models/subscription.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { array, boolean, lazy, object, optional, string, } from '../schema.js';
import { linkDescriptionSchema } from './linkDescription.js';
import { moneySchema } from './money.js';
import { planDetailsSchema } from './planDetails.js';
import { subscriberSchema } from './subscriber.js';
import { subscriptionBillingInformationSchema, } from './subscriptionBillingInformation.js';
export const subscriptionSchema = object({
    id: ['id', optional(string())],
    planId: ['plan_id', optional(string())],
    startTime: ['start_time', optional(string())],
    quantity: ['quantity', optional(string())],
    shippingAmount: ['shipping_amount', optional(lazy(() => moneySchema))],
    subscriber: ['subscriber', optional(lazy(() => subscriberSchema))],
    billingInfo: [
        'billing_info',
        optional(lazy(() => subscriptionBillingInformationSchema)),
    ],
    createTime: ['create_time', optional(string())],
    updateTime: ['update_time', optional(string())],
    customId: ['custom_id', optional(string())],
    planOverridden: ['plan_overridden', optional(boolean())],
    plan: ['plan', optional(lazy(() => planDetailsSchema))],
    links: ['links', optional(array(lazy(() => linkDescriptionSchema)))],
});
