// @paypal/paypal-server-sdk/dist/esm/models/vaultedDigitalWallet.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, lazy, object, optional, string } from '../schema.js';
import { paypalPaymentTokenCustomerTypeSchema, } from './paypalPaymentTokenCustomerType.js';
import { paypalPaymentTokenUsageTypeSchema, } from './paypalPaymentTokenUsageType.js';
import { usagePatternSchema } from './usagePattern.js';
import { vaultedDigitalWalletShippingDetailsSchema, } from './vaultedDigitalWalletShippingDetails.js';
export const vaultedDigitalWalletSchema = object({
    description: ['description', optional(string())],
    usagePattern: ['usage_pattern', optional(usagePatternSchema)],
    shipping: [
        'shipping',
        optional(lazy(() => vaultedDigitalWalletShippingDetailsSchema)),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        optional(boolean()),
    ],
    usageType: ['usage_type', optional(paypalPaymentTokenUsageTypeSchema)],
    customerType: [
        'customer_type',
        optional(paypalPaymentTokenCustomerTypeSchema),
    ],
});
