// @paypal/paypal-server-sdk/dist/esm/models/vaultExperienceContext.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { appSwitchContextSchema, } from './appSwitchContext.js';
import { experienceContextShippingPreferenceSchema, } from './experienceContextShippingPreference.js';
import { vaultInstructionActionSchema, } from './vaultInstructionAction.js';
import { vaultUserActionSchema } from './vaultUserAction.js';
export const vaultExperienceContextSchema = object({
    brandName: ['brand_name', optional(string())],
    locale: ['locale', optional(string())],
    returnUrl: ['return_url', optional(string())],
    cancelUrl: ['cancel_url', optional(string())],
    shippingPreference: [
        'shipping_preference',
        optional(experienceContextShippingPreferenceSchema),
    ],
    vaultInstruction: [
        'vault_instruction',
        optional(vaultInstructionActionSchema),
    ],
    appSwitchContext: [
        'app_switch_context',
        optional(lazy(() => appSwitchContextSchema)),
    ],
    userAction: ['user_action', optional(vaultUserActionSchema)],
});
