// @paypal/paypal-server-sdk/dist/esm/models/venmoWalletVaultAttributes.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, object, optional, string } from '../schema.js';
import { storeInVaultInstructionSchema, } from './storeInVaultInstruction.js';
import { venmoPaymentTokenCustomerTypeSchema, } from './venmoPaymentTokenCustomerType.js';
import { venmoPaymentTokenUsagePatternSchema, } from './venmoPaymentTokenUsagePattern.js';
import { venmoPaymentTokenUsageTypeSchema, } from './venmoPaymentTokenUsageType.js';
export const venmoWalletVaultAttributesSchema = object({
    storeInVault: ['store_in_vault', storeInVaultInstructionSchema],
    description: ['description', optional(string())],
    usagePattern: [
        'usage_pattern',
        optional(venmoPaymentTokenUsagePatternSchema),
    ],
    usageType: ['usage_type', venmoPaymentTokenUsageTypeSchema],
    customerType: [
        'customer_type',
        optional(venmoPaymentTokenCustomerTypeSchema),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        optional(boolean()),
    ],
});
