// @paypal/paypal-server-sdk/dist/esm/models/venmoPaymentToken.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { boolean, lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { nameSchema } from './name.js';
import { paypalPaymentTokenCustomerTypeSchema, } from './paypalPaymentTokenCustomerType.js';
import { paypalPaymentTokenUsageTypeSchema, } from './paypalPaymentTokenUsageType.js';
import { phoneWithTypeSchema } from './phoneWithType.js';
import { usagePatternSchema } from './usagePattern.js';
import { vaultedDigitalWalletShippingDetailsSchema, } from './vaultedDigitalWalletShippingDetails.js';
export const venmoPaymentTokenSchema = object({
    description: ['description', optional(string())],
    usagePattern: ['usage_pattern', optional(usagePatternSchema)],
    shipping: [
        'shipping',
        optional(lazy(() => vaultedDigitalWalletShippingDetailsSchema)),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        optional(boolean()),
    ],
    usageType: ['usage_type', optional(paypalPaymentTokenUsageTypeSchema)],
    customerType: [
        'customer_type',
        optional(paypalPaymentTokenCustomerTypeSchema),
    ],
    emailAddress: ['email_address', optional(string())],
    payerId: ['payer_id', optional(string())],
    name: ['name', optional(lazy(() => nameSchema))],
    phone: ['phone', optional(lazy(() => phoneWithTypeSchema))],
    address: ['address', optional(lazy(() => addressSchema))],
    userName: ['user_name', optional(string())],
});
