// @paypal/paypal-server-sdk/dist/esm/models/venmoWalletResponse.js
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
import { lazy, object, optional, string } from '../schema.js';
import { addressSchema } from './address.js';
import { nameSchema } from './name.js';
import { phoneNumberSchema } from './phoneNumber.js';
import { returnFlowSchema } from './returnFlow.js';
import { venmoWalletAttributesResponseSchema, } from './venmoWalletAttributesResponse.js';
export const venmoWalletResponseSchema = object({
    emailAddress: ['email_address', optional(string())],
    accountId: ['account_id', optional(string())],
    userName: ['user_name', optional(string())],
    name: ['name', optional(lazy(() => nameSchema))],
    phoneNumber: ['phone_number', optional(lazy(() => phoneNumberSchema))],
    address: ['address', optional(lazy(() => addressSchema))],
    returnFlow: ['return_flow', optional(returnFlowSchema)],
    attributes: [
        'attributes',
        optional(lazy(() => venmoWalletAttributesResponseSchema)),
    ],
});
