// @paypal/paypal-server-sdk/dist/cjs/models/applePayPaymentObject.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.applePayPaymentObjectSchema = void 0;
var schema_js_1 = require("../schema.js");
var applePayAttributesResponse_js_1 = require("./applePayAttributesResponse.js");
var applePayCardResponse_js_1 = require("./applePayCardResponse.js");
var cardStoredCredential_js_1 = require("./cardStoredCredential.js");
var phoneNumber_js_1 = require("./phoneNumber.js");
exports.applePayPaymentObjectSchema = (0, schema_js_1.object)({
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    token: ['token', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    phoneNumber: ['phone_number', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumber_js_1.phoneNumberSchema; }))],
    card: ['card', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayCardResponse_js_1.applePayCardResponseSchema; }))],
    attributes: [
        'attributes',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayAttributesResponse_js_1.applePayAttributesResponseSchema; })),
    ],
    storedCredential: [
        'stored_credential',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardStoredCredential_js_1.cardStoredCredentialSchema; })),
    ],
});
