// @paypal/paypal-server-sdk/dist/cjs/models/applePayRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.applePayRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var applePayAttributes_js_1 = require("./applePayAttributes.js");
var applePayDecryptedTokenData_js_1 = require("./applePayDecryptedTokenData.js");
var applePayExperienceContext_js_1 = require("./applePayExperienceContext.js");
var cardStoredCredential_js_1 = require("./cardStoredCredential.js");
var phoneNumber_js_1 = require("./phoneNumber.js");
exports.applePayRequestSchema = (0, schema_js_1.object)({
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    phoneNumber: ['phone_number', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumber_js_1.phoneNumberSchema; }))],
    decryptedToken: [
        'decrypted_token',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayDecryptedTokenData_js_1.applePayDecryptedTokenDataSchema; })),
    ],
    storedCredential: [
        'stored_credential',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardStoredCredential_js_1.cardStoredCredentialSchema; })),
    ],
    vaultId: ['vault_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    attributes: ['attributes', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayAttributes_js_1.applePayAttributesSchema; }))],
    experienceContext: [
        'experience_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayExperienceContext_js_1.applePayExperienceContextSchema; })),
    ],
});
