// @paypal/paypal-server-sdk/dist/cjs/models/applePayDecryptedTokenData.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.applePayDecryptedTokenDataSchema = void 0;
var schema_js_1 = require("../schema.js");
var applePayPaymentData_js_1 = require("./applePayPaymentData.js");
var applePayPaymentDataType_js_1 = require("./applePayPaymentDataType.js");
var applePayTokenizedCard_js_1 = require("./applePayTokenizedCard.js");
var money_js_1 = require("./money.js");
exports.applePayDecryptedTokenDataSchema = (0, schema_js_1.object)({
    transactionAmount: [
        'transaction_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    tokenizedCard: ['tokenized_card', (0, schema_js_1.lazy)(function () { return applePayTokenizedCard_js_1.applePayTokenizedCardSchema; })],
    deviceManufacturerId: ['device_manufacturer_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paymentDataType: [
        'payment_data_type',
        (0, schema_js_1.optional)(applePayPaymentDataType_js_1.applePayPaymentDataTypeSchema),
    ],
    paymentData: [
        'payment_data',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayPaymentData_js_1.applePayPaymentDataSchema; })),
    ],
});
