// @paypal/paypal-server-sdk/dist/cjs/models/applePayCardResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.applePayCardResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var authenticationResponse_js_1 = require("./authenticationResponse.js");
var binDetails_js_1 = require("./binDetails.js");
var cardAttributesResponse_js_1 = require("./cardAttributesResponse.js");
var cardBrand_js_1 = require("./cardBrand.js");
var cardFromRequest_js_1 = require("./cardFromRequest.js");
var cardStoredCredential_js_1 = require("./cardStoredCredential.js");
var cardType_js_1 = require("./cardType.js");
exports.applePayCardResponseSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    lastDigits: ['last_digits', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    brand: ['brand', (0, schema_js_1.optional)(cardBrand_js_1.cardBrandSchema)],
    availableNetworks: ['available_networks', (0, schema_js_1.optional)((0, schema_js_1.array)(cardBrand_js_1.cardBrandSchema))],
    type: ['type', (0, schema_js_1.optional)(cardType_js_1.cardTypeSchema)],
    authenticationResult: [
        'authentication_result',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return authenticationResponse_js_1.authenticationResponseSchema; })),
    ],
    attributes: [
        'attributes',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardAttributesResponse_js_1.cardAttributesResponseSchema; })),
    ],
    fromRequest: ['from_request', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardFromRequest_js_1.cardFromRequestSchema; }))],
    expiry: ['expiry', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    binDetails: ['bin_details', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return binDetails_js_1.binDetailsSchema; }))],
    storedCredential: [
        'stored_credential',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardStoredCredential_js_1.cardStoredCredentialSchema; })),
    ],
    billingAddress: ['billing_address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    countryCode: ['country_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
