// @paypal/paypal-server-sdk/dist/cjs/models/billingPlan.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.billingPlanSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var merchantPreferences_js_1 = require("./merchantPreferences.js");
var paymentPreferences_js_1 = require("./paymentPreferences.js");
var subscriptionBillingCycle_js_1 = require("./subscriptionBillingCycle.js");
var subscriptionPlanStatus_js_1 = require("./subscriptionPlanStatus.js");
var taxes_js_1 = require("./taxes.js");
exports.billingPlanSchema = (0, schema_js_1.object)({
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    productId: ['product_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    status: ['status', (0, schema_js_1.optional)(subscriptionPlanStatus_js_1.subscriptionPlanStatusSchema)],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    billingCycles: [
        'billing_cycles',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return subscriptionBillingCycle_js_1.subscriptionBillingCycleSchema; }))),
    ],
    paymentPreferences: [
        'payment_preferences',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentPreferences_js_1.paymentPreferencesSchema; })),
    ],
    merchantPreferences: [
        'merchant_preferences',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return merchantPreferences_js_1.merchantPreferencesSchema; })),
    ],
    taxes: ['taxes', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return taxes_js_1.taxesSchema; }))],
    quantitySupported: ['quantity_supported', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
