// @paypal/paypal-server-sdk/dist/cjs/models/blikPaymentRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.blikPaymentRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var blikExperienceContext_js_1 = require("./blikExperienceContext.js");
var blikLevel0PaymentObject_js_1 = require("./blikLevel0PaymentObject.js");
var blikOneClickPaymentRequest_js_1 = require("./blikOneClickPaymentRequest.js");
exports.blikPaymentRequestSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.string)()],
    countryCode: ['country_code', (0, schema_js_1.string)()],
    email: ['email', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    experienceContext: [
        'experience_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return blikExperienceContext_js_1.blikExperienceContextSchema; })),
    ],
    level0: ['level_0', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return blikLevel0PaymentObject_js_1.blikLevel0PaymentObjectSchema; }))],
    oneClick: [
        'one_click',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return blikOneClickPaymentRequest_js_1.blikOneClickPaymentRequestSchema; })),
    ],
});
