// @paypal/paypal-server-sdk/dist/cjs/models/captureRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.captureRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var capturePaymentInstruction_js_1 = require("./capturePaymentInstruction.js");
var money_js_1 = require("./money.js");
exports.captureRequestSchema = (0, schema_js_1.object)({
    amount: ['amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    finalCapture: ['final_capture', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    paymentInstruction: [
        'payment_instruction',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return capturePaymentInstruction_js_1.capturePaymentInstructionSchema; })),
    ],
    noteToPayer: ['note_to_payer', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    softDescriptor: ['soft_descriptor', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
