// @paypal/paypal-server-sdk/dist/cjs/models/cardVerificationDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.cardVerificationDetailsSchema = void 0;
var schema_js_1 = require("../schema.js");
var cardBrand_js_1 = require("./cardBrand.js");
var cardVerificationProcessorResponse_js_1 = require("./cardVerificationProcessorResponse.js");
var money_js_1 = require("./money.js");
exports.cardVerificationDetailsSchema = (0, schema_js_1.object)({
    networkTransactionId: ['network_transaction_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    date: ['date', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    network: ['network', (0, schema_js_1.optional)(cardBrand_js_1.cardBrandSchema)],
    time: ['time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    amount: ['amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    processorResponse: [
        'processor_response',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardVerificationProcessorResponse_js_1.cardVerificationProcessorResponseSchema; })),
    ],
    threeDSecure: ['three_d_secure', (0, schema_js_1.optional)((0, schema_js_1.unknown)())],
});
