// @paypal/paypal-server-sdk/dist/cjs/models/cardPaymentTokenEntity.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.cardPaymentTokenEntitySchema = void 0;
var schema_js_1 = require("../schema.js");
var binDetails_js_1 = require("./binDetails.js");
var cardAuthenticationResponse_js_1 = require("./cardAuthenticationResponse.js");
var cardBrand_js_1 = require("./cardBrand.js");
var cardResponseAddress_js_1 = require("./cardResponseAddress.js");
var cardType_js_1 = require("./cardType.js");
var cardVerificationDetails_js_1 = require("./cardVerificationDetails.js");
var cardVerificationStatus_js_1 = require("./cardVerificationStatus.js");
var networkTransactionReferenceEntity_js_1 = require("./networkTransactionReferenceEntity.js");
exports.cardPaymentTokenEntitySchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    lastDigits: ['last_digits', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    brand: ['brand', (0, schema_js_1.optional)(cardBrand_js_1.cardBrandSchema)],
    expiry: ['expiry', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    billingAddress: [
        'billing_address',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardResponseAddress_js_1.cardResponseAddressSchema; })),
    ],
    verificationStatus: [
        'verification_status',
        (0, schema_js_1.optional)(cardVerificationStatus_js_1.cardVerificationStatusSchema),
    ],
    verification: [
        'verification',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardVerificationDetails_js_1.cardVerificationDetailsSchema; })),
    ],
    networkTransactionReference: [
        'network_transaction_reference',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return networkTransactionReferenceEntity_js_1.networkTransactionReferenceEntitySchema; })),
    ],
    authenticationResult: [
        'authentication_result',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardAuthenticationResponse_js_1.cardAuthenticationResponseSchema; })),
    ],
    binDetails: ['bin_details', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return binDetails_js_1.binDetailsSchema; }))],
    type: ['type', (0, schema_js_1.optional)(cardType_js_1.cardTypeSchema)],
});
