// @paypal/paypal-server-sdk/dist/cjs/models/cardRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.cardRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var cardAttributes_js_1 = require("./cardAttributes.js");
var cardExperienceContext_js_1 = require("./cardExperienceContext.js");
var cardStoredCredential_js_1 = require("./cardStoredCredential.js");
var networkToken_js_1 = require("./networkToken.js");
exports.cardRequestSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    number: ['number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    expiry: ['expiry', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    securityCode: ['security_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    billingAddress: ['billing_address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    attributes: ['attributes', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardAttributes_js_1.cardAttributesSchema; }))],
    vaultId: ['vault_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    singleUseToken: ['single_use_token', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    storedCredential: [
        'stored_credential',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardStoredCredential_js_1.cardStoredCredentialSchema; })),
    ],
    networkToken: ['network_token', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return networkToken_js_1.networkTokenSchema; }))],
    experienceContext: [
        'experience_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardExperienceContext_js_1.cardExperienceContextSchema; })),
    ],
});
