// @paypal/paypal-server-sdk/dist/cjs/models/createSubscriptionRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createSubscriptionRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
var planOverride_js_1 = require("./planOverride.js");
var subscriberRequest_js_1 = require("./subscriberRequest.js");
var subscriptionApplicationContext_js_1 = require("./subscriptionApplicationContext.js");
exports.createSubscriptionRequestSchema = (0, schema_js_1.object)({
    planId: ['plan_id', (0, schema_js_1.string)()],
    startTime: ['start_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    quantity: ['quantity', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    shippingAmount: ['shipping_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    subscriber: ['subscriber', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriberRequest_js_1.subscriberRequestSchema; }))],
    autoRenewal: ['auto_renewal', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    applicationContext: [
        'application_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriptionApplicationContext_js_1.subscriptionApplicationContextSchema; })),
    ],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    plan: ['plan', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return planOverride_js_1.planOverrideSchema; }))],
});
