// @paypal/paypal-server-sdk/dist/cjs/models/customerVaultPaymentTokensResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.customerVaultPaymentTokensResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var paymentTokenResponse_js_1 = require("./paymentTokenResponse.js");
var vaultResponseCustomer_js_1 = require("./vaultResponseCustomer.js");
exports.customerVaultPaymentTokensResponseSchema = (0, schema_js_1.object)({
    totalItems: ['total_items', (0, schema_js_1.optional)((0, schema_js_1.number)())],
    totalPages: ['total_pages', (0, schema_js_1.optional)((0, schema_js_1.number)())],
    customer: ['customer', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultResponseCustomer_js_1.vaultResponseCustomerSchema; }))],
    paymentTokens: [
        'payment_tokens',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return paymentTokenResponse_js_1.paymentTokenResponseSchema; }))),
    ],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
