// @paypal/paypal-server-sdk/dist/cjs/models/googlePayCardResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.googlePayCardResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var authenticationResponse_js_1 = require("./authenticationResponse.js");
var cardBrand_js_1 = require("./cardBrand.js");
var cardType_js_1 = require("./cardType.js");
exports.googlePayCardResponseSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    lastDigits: ['last_digits', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    type: ['type', (0, schema_js_1.optional)(cardType_js_1.cardTypeSchema)],
    brand: ['brand', (0, schema_js_1.optional)(cardBrand_js_1.cardBrandSchema)],
    billingAddress: ['billing_address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    authenticationResult: [
        'authentication_result',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return authenticationResponse_js_1.authenticationResponseSchema; })),
    ],
});
