// @paypal/paypal-server-sdk/dist/cjs/models/googlePayRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.googlePayRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var assuranceDetails_js_1 = require("./assuranceDetails.js");
var googlePayDecryptedTokenData_js_1 = require("./googlePayDecryptedTokenData.js");
var googlePayExperienceContext_js_1 = require("./googlePayExperienceContext.js");
var googlePayRequestCard_js_1 = require("./googlePayRequestCard.js");
var phoneNumberWithCountryCode_js_1 = require("./phoneNumberWithCountryCode.js");
exports.googlePayRequestSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    phoneNumber: [
        'phone_number',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumberWithCountryCode_js_1.phoneNumberWithCountryCodeSchema; })),
    ],
    card: ['card', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return googlePayRequestCard_js_1.googlePayRequestCardSchema; }))],
    decryptedToken: [
        'decrypted_token',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return googlePayDecryptedTokenData_js_1.googlePayDecryptedTokenDataSchema; })),
    ],
    assuranceDetails: [
        'assurance_details',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return assuranceDetails_js_1.assuranceDetailsSchema; })),
    ],
    experienceContext: [
        'experience_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return googlePayExperienceContext_js_1.googlePayExperienceContextSchema; })),
    ],
});
