// @paypal/paypal-server-sdk/dist/cjs/models/googlePayDecryptedTokenData.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.googlePayDecryptedTokenDataSchema = void 0;
var schema_js_1 = require("../schema.js");
var googlePayAuthenticationMethod_js_1 = require("./googlePayAuthenticationMethod.js");
var googlePayCard_js_1 = require("./googlePayCard.js");
var googlePayPaymentMethod_js_1 = require("./googlePayPaymentMethod.js");
exports.googlePayDecryptedTokenDataSchema = (0, schema_js_1.object)({
    messageId: ['message_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    messageExpiration: ['message_expiration', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paymentMethod: ['payment_method', googlePayPaymentMethod_js_1.googlePayPaymentMethodSchema],
    card: ['card', (0, schema_js_1.lazy)(function () { return googlePayCard_js_1.googlePayCardSchema; })],
    authenticationMethod: [
        'authentication_method',
        googlePayAuthenticationMethod_js_1.googlePayAuthenticationMethodSchema,
    ],
    cryptogram: ['cryptogram', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    eciIndicator: ['eci_indicator', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
