// @paypal/paypal-server-sdk/dist/cjs/models/itemDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.itemDetailsSchema = void 0;
var schema_js_1 = require("../schema.js");
var checkoutOption_js_1 = require("./checkoutOption.js");
var money_js_1 = require("./money.js");
var taxAmount_js_1 = require("./taxAmount.js");
exports.itemDetailsSchema = (0, schema_js_1.object)({
    itemCode: ['item_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    itemName: ['item_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    itemDescription: ['item_description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    itemOptions: ['item_options', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    itemQuantity: ['item_quantity', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    itemUnitPrice: ['item_unit_price', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    itemAmount: ['item_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    discountAmount: ['discount_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    adjustmentAmount: ['adjustment_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    giftWrapAmount: ['gift_wrap_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    taxPercentage: ['tax_percentage', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    taxAmounts: ['tax_amounts', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return taxAmount_js_1.taxAmountSchema; })))],
    basicShippingAmount: [
        'basic_shipping_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    extraShippingAmount: [
        'extra_shipping_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    handlingAmount: ['handling_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    insuranceAmount: ['insurance_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    totalItemAmount: ['total_item_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    invoiceNumber: ['invoice_number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    checkoutOptions: [
        'checkout_options',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return checkoutOption_js_1.checkoutOptionSchema; }))),
    ],
});
