// @paypal/paypal-server-sdk/dist/cjs/models/level3CardProcessingData.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.level3CardProcessingDataSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var lineItem_js_1 = require("./lineItem.js");
var money_js_1 = require("./money.js");
exports.level3CardProcessingDataSchema = (0, schema_js_1.object)({
    shippingAmount: ['shipping_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    dutyAmount: ['duty_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    discountAmount: ['discount_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    shippingAddress: ['shipping_address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    shipsFromPostalCode: ['ships_from_postal_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    lineItems: ['line_items', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return lineItem_js_1.lineItemSchema; })))],
});
