// @paypal/paypal-server-sdk/dist/cjs/models/lineItem.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.lineItemSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
var orderBillingPlan_js_1 = require("./orderBillingPlan.js");
var universalProductCode_js_1 = require("./universalProductCode.js");
exports.lineItemSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.string)()],
    quantity: ['quantity', (0, schema_js_1.string)()],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    sku: ['sku', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    url: ['url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    imageUrl: ['image_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    upc: ['upc', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return universalProductCode_js_1.universalProductCodeSchema; }))],
    billingPlan: ['billing_plan', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return orderBillingPlan_js_1.orderBillingPlanSchema; }))],
    unitAmount: ['unit_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    tax: ['tax', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    commodityCode: ['commodity_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    discountAmount: ['discount_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    totalAmount: ['total_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    unitOfMeasure: ['unit_of_measure', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
