// @paypal/paypal-server-sdk/dist/cjs/models/modifySubscriptionResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.modifySubscriptionResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var money_js_1 = require("./money.js");
var planOverride_js_1 = require("./planOverride.js");
var shippingDetails_js_1 = require("./shippingDetails.js");
exports.modifySubscriptionResponseSchema = (0, schema_js_1.object)({
    planId: ['plan_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    quantity: ['quantity', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    shippingAmount: ['shipping_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    shippingAddress: [
        'shipping_address',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return shippingDetails_js_1.shippingDetailsSchema; })),
    ],
    plan: ['plan', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return planOverride_js_1.planOverrideSchema; }))],
    planOverridden: ['plan_overridden', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
