// @paypal/paypal-server-sdk/dist/cjs/models/oAuthProviderError.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.oAuthProviderErrorSchema = exports.OAuthProviderError = void 0;
var schema_js_1 = require("../schema.js");
/**
 * Enum for OAuthProviderError
 */
var OAuthProviderError;
(function (OAuthProviderError) {
    OAuthProviderError["InvalidRequest"] = "invalid_request";
    OAuthProviderError["InvalidClient"] = "invalid_client";
    OAuthProviderError["InvalidGrant"] = "invalid_grant";
    OAuthProviderError["UnauthorizedClient"] = "unauthorized_client";
    OAuthProviderError["UnsupportedGrantType"] = "unsupported_grant_type";
    OAuthProviderError["InvalidScope"] = "invalid_scope";
})(OAuthProviderError || (exports.OAuthProviderError = OAuthProviderError = {}));
/**
 * Schema for OAuthProviderError
 */
exports.oAuthProviderErrorSchema = (0, schema_js_1.stringEnum)(OAuthProviderError);
