// @paypal/paypal-server-sdk/dist/cjs/models/orderApplicationContext.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderApplicationContextSchema = void 0;
var schema_js_1 = require("../schema.js");
var orderApplicationContextLandingPage_js_1 = require("./orderApplicationContextLandingPage.js");
var orderApplicationContextShippingPreference_js_1 = require("./orderApplicationContextShippingPreference.js");
var orderApplicationContextUserAction_js_1 = require("./orderApplicationContextUserAction.js");
var paymentMethodPreference_js_1 = require("./paymentMethodPreference.js");
var storedPaymentSource_js_1 = require("./storedPaymentSource.js");
exports.orderApplicationContextSchema = (0, schema_js_1.object)({
    brandName: ['brand_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    locale: ['locale', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    landingPage: [
        'landing_page',
        (0, schema_js_1.optional)(orderApplicationContextLandingPage_js_1.orderApplicationContextLandingPageSchema),
    ],
    shippingPreference: [
        'shipping_preference',
        (0, schema_js_1.optional)(orderApplicationContextShippingPreference_js_1.orderApplicationContextShippingPreferenceSchema),
    ],
    userAction: [
        'user_action',
        (0, schema_js_1.optional)(orderApplicationContextUserAction_js_1.orderApplicationContextUserActionSchema),
    ],
    paymentMethod: [
        'payment_method',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentMethodPreference_js_1.paymentMethodPreferenceSchema; })),
    ],
    returnUrl: ['return_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    cancelUrl: ['cancel_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    storedPaymentSource: [
        'stored_payment_source',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return storedPaymentSource_js_1.storedPaymentSourceSchema; })),
    ],
});
