// @paypal/paypal-server-sdk/dist/cjs/models/orderAuthorizeResponsePaymentSource.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderAuthorizeResponsePaymentSourceSchema = void 0;
var schema_js_1 = require("../schema.js");
var applePayPaymentObject_js_1 = require("./applePayPaymentObject.js");
var cardResponse_js_1 = require("./cardResponse.js");
var googlePayWalletResponse_js_1 = require("./googlePayWalletResponse.js");
var paypalWalletResponse_js_1 = require("./paypalWalletResponse.js");
var venmoWalletResponse_js_1 = require("./venmoWalletResponse.js");
exports.orderAuthorizeResponsePaymentSourceSchema = (0, schema_js_1.object)({
    card: ['card', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardResponse_js_1.cardResponseSchema; }))],
    paypal: ['paypal', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paypalWalletResponse_js_1.paypalWalletResponseSchema; }))],
    applePay: ['apple_pay', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayPaymentObject_js_1.applePayPaymentObjectSchema; }))],
    googlePay: [
        'google_pay',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return googlePayWalletResponse_js_1.googlePayWalletResponseSchema; })),
    ],
    venmo: ['venmo', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return venmoWalletResponse_js_1.venmoWalletResponseSchema; }))],
});
