// @paypal/paypal-server-sdk/dist/cjs/models/orderAuthorizeRequestPaymentSource.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderAuthorizeRequestPaymentSourceSchema = void 0;
var schema_js_1 = require("../schema.js");
var applePayRequest_js_1 = require("./applePayRequest.js");
var cardRequest_js_1 = require("./cardRequest.js");
var googlePayRequest_js_1 = require("./googlePayRequest.js");
var paypalWallet_js_1 = require("./paypalWallet.js");
var token_js_1 = require("./token.js");
var venmoWalletRequest_js_1 = require("./venmoWalletRequest.js");
exports.orderAuthorizeRequestPaymentSourceSchema = (0, schema_js_1.object)({
    card: ['card', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return cardRequest_js_1.cardRequestSchema; }))],
    token: ['token', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return token_js_1.tokenSchema; }))],
    paypal: ['paypal', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paypalWallet_js_1.paypalWalletSchema; }))],
    applePay: ['apple_pay', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return applePayRequest_js_1.applePayRequestSchema; }))],
    googlePay: ['google_pay', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return googlePayRequest_js_1.googlePayRequestSchema; }))],
    venmo: ['venmo', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return venmoWalletRequest_js_1.venmoWalletRequestSchema; }))],
});
