// @paypal/paypal-server-sdk/dist/cjs/models/orderAuthorizeResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderAuthorizeResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var checkoutPaymentIntent_js_1 = require("./checkoutPaymentIntent.js");
var linkDescription_js_1 = require("./linkDescription.js");
var orderAuthorizeResponsePaymentSource_js_1 = require("./orderAuthorizeResponsePaymentSource.js");
var orderStatus_js_1 = require("./orderStatus.js");
var payer_js_1 = require("./payer.js");
var purchaseUnit_js_1 = require("./purchaseUnit.js");
exports.orderAuthorizeResponseSchema = (0, schema_js_1.object)({
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paymentSource: [
        'payment_source',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return orderAuthorizeResponsePaymentSource_js_1.orderAuthorizeResponsePaymentSourceSchema; })),
    ],
    intent: ['intent', (0, schema_js_1.optional)(checkoutPaymentIntent_js_1.checkoutPaymentIntentSchema)],
    payer: ['payer', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return payer_js_1.payerSchema; }))],
    purchaseUnits: [
        'purchase_units',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return purchaseUnit_js_1.purchaseUnitSchema; }))),
    ],
    status: ['status', (0, schema_js_1.optional)(orderStatus_js_1.orderStatusSchema)],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
