// @paypal/paypal-server-sdk/dist/cjs/models/ordersCapture.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ordersCaptureSchema = void 0;
var schema_js_1 = require("../schema.js");
var captureStatus_js_1 = require("./captureStatus.js");
var captureStatusDetails_js_1 = require("./captureStatusDetails.js");
var disbursementMode_js_1 = require("./disbursementMode.js");
var linkDescription_js_1 = require("./linkDescription.js");
var money_js_1 = require("./money.js");
var networkTransaction_js_1 = require("./networkTransaction.js");
var processorResponse_js_1 = require("./processorResponse.js");
var sellerProtection_js_1 = require("./sellerProtection.js");
var sellerReceivableBreakdown_js_1 = require("./sellerReceivableBreakdown.js");
exports.ordersCaptureSchema = (0, schema_js_1.object)({
    status: ['status', (0, schema_js_1.optional)(captureStatus_js_1.captureStatusSchema)],
    statusDetails: [
        'status_details',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return captureStatusDetails_js_1.captureStatusDetailsSchema; })),
    ],
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    amount: ['amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    networkTransactionReference: [
        'network_transaction_reference',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return networkTransaction_js_1.networkTransactionSchema; })),
    ],
    sellerProtection: [
        'seller_protection',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return sellerProtection_js_1.sellerProtectionSchema; })),
    ],
    finalCapture: ['final_capture', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    sellerReceivableBreakdown: [
        'seller_receivable_breakdown',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return sellerReceivableBreakdown_js_1.sellerReceivableBreakdownSchema; })),
    ],
    disbursementMode: ['disbursement_mode', (0, schema_js_1.optional)(disbursementMode_js_1.disbursementModeSchema)],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
    processorResponse: [
        'processor_response',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return processorResponse_js_1.processorResponseSchema; })),
    ],
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
