// @paypal/paypal-server-sdk/dist/cjs/models/orderTrackerRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderTrackerRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var orderTrackerItem_js_1 = require("./orderTrackerItem.js");
var shipmentCarrier_js_1 = require("./shipmentCarrier.js");
exports.orderTrackerRequestSchema = (0, schema_js_1.object)({
    trackingNumber: ['tracking_number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    carrier: ['carrier', (0, schema_js_1.optional)(shipmentCarrier_js_1.shipmentCarrierSchema)],
    carrierNameOther: ['carrier_name_other', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    captureId: ['capture_id', (0, schema_js_1.string)()],
    notifyPayer: ['notify_payer', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    items: ['items', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return orderTrackerItem_js_1.orderTrackerItemSchema; })))],
});
