// @paypal/paypal-server-sdk/dist/cjs/models/orderTrackerResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderTrackerResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var orderTrackerItem_js_1 = require("./orderTrackerItem.js");
var orderTrackerStatus_js_1 = require("./orderTrackerStatus.js");
exports.orderTrackerResponseSchema = (0, schema_js_1.object)({
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    status: ['status', (0, schema_js_1.optional)(orderTrackerStatus_js_1.orderTrackerStatusSchema)],
    items: ['items', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return orderTrackerItem_js_1.orderTrackerItemSchema; })))],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
