// @paypal/paypal-server-sdk/dist/cjs/models/orderRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var checkoutPaymentIntent_js_1 = require("./checkoutPaymentIntent.js");
var orderApplicationContext_js_1 = require("./orderApplicationContext.js");
var payer_js_1 = require("./payer.js");
var paymentSource_js_1 = require("./paymentSource.js");
var purchaseUnitRequest_js_1 = require("./purchaseUnitRequest.js");
exports.orderRequestSchema = (0, schema_js_1.object)({
    intent: ['intent', checkoutPaymentIntent_js_1.checkoutPaymentIntentSchema],
    payer: ['payer', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return payer_js_1.payerSchema; }))],
    purchaseUnits: [
        'purchase_units',
        (0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return purchaseUnitRequest_js_1.purchaseUnitRequestSchema; })),
    ],
    paymentSource: ['payment_source', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentSource_js_1.paymentSourceSchema; }))],
    applicationContext: [
        'application_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return orderApplicationContext_js_1.orderApplicationContextSchema; })),
    ],
});
