// @paypal/paypal-server-sdk/dist/cjs/models/payerInformation.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.payerInformationSchema = void 0;
var schema_js_1 = require("../schema.js");
var payerName_js_1 = require("./payerName.js");
var phone_js_1 = require("./phone.js");
var simplePostalAddressCoarseGrained_js_1 = require("./simplePostalAddressCoarseGrained.js");
exports.payerInformationSchema = (0, schema_js_1.object)({
    accountId: ['account_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    phoneNumber: ['phone_number', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phone_js_1.phoneSchema; }))],
    addressStatus: ['address_status', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    payerStatus: ['payer_status', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    payerName: ['payer_name', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return payerName_js_1.payerNameSchema; }))],
    countryCode: ['country_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    address: [
        'address',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return simplePostalAddressCoarseGrained_js_1.simplePostalAddressCoarseGrainedSchema; })),
    ],
});
