// @paypal/paypal-server-sdk/dist/cjs/models/paymentAdviceCode.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.paymentAdviceCodeSchema = exports.PaymentAdviceCode = void 0;
var schema_js_1 = require("../schema.js");
/**
 * Enum for PaymentAdviceCode
 */
var PaymentAdviceCode;
(function (PaymentAdviceCode) {
    PaymentAdviceCode["PaymentAdvice01"] = "01";
    PaymentAdviceCode["PaymentAdvice02"] = "02";
    PaymentAdviceCode["PaymentAdvice03"] = "03";
    PaymentAdviceCode["PaymentAdvice04"] = "04";
    PaymentAdviceCode["PaymentAdvice21"] = "21";
    PaymentAdviceCode["PaymentAdvice22"] = "22";
    PaymentAdviceCode["PaymentAdvice24"] = "24";
    PaymentAdviceCode["PaymentAdvice25"] = "25";
    PaymentAdviceCode["PaymentAdvice26"] = "26";
    PaymentAdviceCode["PaymentAdvice27"] = "27";
    PaymentAdviceCode["PaymentAdvice28"] = "28";
    PaymentAdviceCode["PaymentAdvice29"] = "29";
    PaymentAdviceCode["PaymentAdvice30"] = "30";
    PaymentAdviceCode["PaymentAdvice40"] = "40";
    PaymentAdviceCode["PaymentAdvice43"] = "43";
})(PaymentAdviceCode || (exports.PaymentAdviceCode = PaymentAdviceCode = {}));
/**
 * Schema for PaymentAdviceCode
 */
exports.paymentAdviceCodeSchema = (0, schema_js_1.stringEnum)(PaymentAdviceCode, true);
