// @paypal/paypal-server-sdk/dist/cjs/models/paypalWalletExperienceContext.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.paypalWalletExperienceContextSchema = void 0;
var schema_js_1 = require("../schema.js");
var appSwitchContext_js_1 = require("./appSwitchContext.js");
var callbackConfiguration_js_1 = require("./callbackConfiguration.js");
var payeePaymentMethodPreference_js_1 = require("./payeePaymentMethodPreference.js");
var paypalExperienceLandingPage_js_1 = require("./paypalExperienceLandingPage.js");
var paypalExperienceUserAction_js_1 = require("./paypalExperienceUserAction.js");
var paypalWalletContactPreference_js_1 = require("./paypalWalletContactPreference.js");
var paypalWalletContextShippingPreference_js_1 = require("./paypalWalletContextShippingPreference.js");
exports.paypalWalletExperienceContextSchema = (0, schema_js_1.object)({
    brandName: ['brand_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    locale: ['locale', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    shippingPreference: [
        'shipping_preference',
        (0, schema_js_1.optional)(paypalWalletContextShippingPreference_js_1.paypalWalletContextShippingPreferenceSchema),
    ],
    contactPreference: [
        'contact_preference',
        (0, schema_js_1.optional)(paypalWalletContactPreference_js_1.paypalWalletContactPreferenceSchema),
    ],
    returnUrl: ['return_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    cancelUrl: ['cancel_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    appSwitchContext: [
        'app_switch_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return appSwitchContext_js_1.appSwitchContextSchema; })),
    ],
    landingPage: ['landing_page', (0, schema_js_1.optional)(paypalExperienceLandingPage_js_1.paypalExperienceLandingPageSchema)],
    userAction: ['user_action', (0, schema_js_1.optional)(paypalExperienceUserAction_js_1.paypalExperienceUserActionSchema)],
    paymentMethodPreference: [
        'payment_method_preference',
        (0, schema_js_1.optional)(payeePaymentMethodPreference_js_1.payeePaymentMethodPreferenceSchema),
    ],
    orderUpdateCallbackConfig: [
        'order_update_callback_config',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return callbackConfiguration_js_1.callbackConfigurationSchema; })),
    ],
});
