// @paypal/paypal-server-sdk/dist/cjs/models/paypalWalletResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.paypalWalletResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var experienceStatus_js_1 = require("./experienceStatus.js");
var name_js_1 = require("./name.js");
var paypalWalletAccountVerificationStatus_js_1 = require("./paypalWalletAccountVerificationStatus.js");
var paypalWalletAttributesResponse_js_1 = require("./paypalWalletAttributesResponse.js");
var paypalWalletStoredCredential_js_1 = require("./paypalWalletStoredCredential.js");
var phoneNumber_js_1 = require("./phoneNumber.js");
var phoneType_js_1 = require("./phoneType.js");
var taxInfo_js_1 = require("./taxInfo.js");
exports.paypalWalletResponseSchema = (0, schema_js_1.object)({
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    accountId: ['account_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    accountStatus: [
        'account_status',
        (0, schema_js_1.optional)(paypalWalletAccountVerificationStatus_js_1.paypalWalletAccountVerificationStatusSchema),
    ],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return name_js_1.nameSchema; }))],
    phoneType: ['phone_type', (0, schema_js_1.optional)(phoneType_js_1.phoneTypeSchema)],
    phoneNumber: ['phone_number', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumber_js_1.phoneNumberSchema; }))],
    birthDate: ['birth_date', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    businessName: ['business_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    taxInfo: ['tax_info', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return taxInfo_js_1.taxInfoSchema; }))],
    address: ['address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    attributes: [
        'attributes',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paypalWalletAttributesResponse_js_1.paypalWalletAttributesResponseSchema; })),
    ],
    storedCredential: [
        'stored_credential',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paypalWalletStoredCredential_js_1.paypalWalletStoredCredentialSchema; })),
    ],
    experienceStatus: ['experience_status', (0, schema_js_1.optional)(experienceStatus_js_1.experienceStatusSchema)],
});
