// @paypal/paypal-server-sdk/dist/cjs/models/planRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.planRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var merchantPreferences_js_1 = require("./merchantPreferences.js");
var paymentPreferences_js_1 = require("./paymentPreferences.js");
var planRequestStatus_js_1 = require("./planRequestStatus.js");
var subscriptionBillingCycle_js_1 = require("./subscriptionBillingCycle.js");
var taxes_js_1 = require("./taxes.js");
exports.planRequestSchema = (0, schema_js_1.object)({
    productId: ['product_id', (0, schema_js_1.string)()],
    name: ['name', (0, schema_js_1.string)()],
    status: ['status', (0, schema_js_1.optional)(planRequestStatus_js_1.planRequestStatusSchema)],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    billingCycles: [
        'billing_cycles',
        (0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return subscriptionBillingCycle_js_1.subscriptionBillingCycleSchema; })),
    ],
    paymentPreferences: [
        'payment_preferences',
        (0, schema_js_1.lazy)(function () { return paymentPreferences_js_1.paymentPreferencesSchema; }),
    ],
    merchantPreferences: [
        'merchant_preferences',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return merchantPreferences_js_1.merchantPreferencesSchema; })),
    ],
    taxes: ['taxes', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return taxes_js_1.taxesSchema; }))],
    quantitySupported: ['quantity_supported', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
});
