// @paypal/paypal-server-sdk/dist/cjs/models/purchaseUnitRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.purchaseUnitRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var amountWithBreakdown_js_1 = require("./amountWithBreakdown.js");
var itemRequest_js_1 = require("./itemRequest.js");
var payeeBase_js_1 = require("./payeeBase.js");
var paymentInstruction_js_1 = require("./paymentInstruction.js");
var shippingDetails_js_1 = require("./shippingDetails.js");
var supplementaryData_js_1 = require("./supplementaryData.js");
exports.purchaseUnitRequestSchema = (0, schema_js_1.object)({
    referenceId: ['reference_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    amount: ['amount', (0, schema_js_1.lazy)(function () { return amountWithBreakdown_js_1.amountWithBreakdownSchema; })],
    payee: ['payee', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return payeeBase_js_1.payeeBaseSchema; }))],
    paymentInstruction: [
        'payment_instruction',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentInstruction_js_1.paymentInstructionSchema; })),
    ],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    softDescriptor: ['soft_descriptor', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    items: ['items', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return itemRequest_js_1.itemRequestSchema; })))],
    shipping: ['shipping', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return shippingDetails_js_1.shippingDetailsSchema; }))],
    supplementaryData: [
        'supplementary_data',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return supplementaryData_js_1.supplementaryDataSchema; })),
    ],
});
