// @paypal/paypal-server-sdk/dist/cjs/models/purchaseUnit.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.purchaseUnitSchema = void 0;
var schema_js_1 = require("../schema.js");
var amountWithBreakdown_js_1 = require("./amountWithBreakdown.js");
var item_js_1 = require("./item.js");
var payeeBase_js_1 = require("./payeeBase.js");
var paymentCollection_js_1 = require("./paymentCollection.js");
var paymentInstruction_js_1 = require("./paymentInstruction.js");
var shippingWithTrackingDetails_js_1 = require("./shippingWithTrackingDetails.js");
var supplementaryData_js_1 = require("./supplementaryData.js");
exports.purchaseUnitSchema = (0, schema_js_1.object)({
    referenceId: ['reference_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    amount: ['amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return amountWithBreakdown_js_1.amountWithBreakdownSchema; }))],
    payee: ['payee', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return payeeBase_js_1.payeeBaseSchema; }))],
    paymentInstruction: [
        'payment_instruction',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentInstruction_js_1.paymentInstructionSchema; })),
    ],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    softDescriptor: ['soft_descriptor', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    items: ['items', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return item_js_1.itemSchema; })))],
    shipping: [
        'shipping',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return shippingWithTrackingDetails_js_1.shippingWithTrackingDetailsSchema; })),
    ],
    supplementaryData: [
        'supplementary_data',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return supplementaryData_js_1.supplementaryDataSchema; })),
    ],
    payments: ['payments', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentCollection_js_1.paymentCollectionSchema; }))],
    mostRecentErrors: ['most_recent_errors', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.unknown)()))],
});
