// @paypal/paypal-server-sdk/dist/cjs/models/reasonCode.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.reasonCodeSchema = exports.ReasonCode = void 0;
var schema_js_1 = require("../schema.js");
/**
 * Enum for ReasonCode
 */
var ReasonCode;
(function (ReasonCode) {
    ReasonCode["PaymentDenied"] = "PAYMENT_DENIED";
    ReasonCode["InternalServerError"] = "INTERNAL_SERVER_ERROR";
    ReasonCode["PayeeAccountRestricted"] = "PAYEE_ACCOUNT_RESTRICTED";
    ReasonCode["PayerAccountRestricted"] = "PAYER_ACCOUNT_RESTRICTED";
    ReasonCode["PayerCannotPay"] = "PAYER_CANNOT_PAY";
    ReasonCode["SendingLimitExceeded"] = "SENDING_LIMIT_EXCEEDED";
    ReasonCode["TransactionReceivingLimitExceeded"] = "TRANSACTION_RECEIVING_LIMIT_EXCEEDED";
    ReasonCode["CurrencyMismatch"] = "CURRENCY_MISMATCH";
})(ReasonCode || (exports.ReasonCode = ReasonCode = {}));
/**
 * Schema for ReasonCode
 */
exports.reasonCodeSchema = (0, schema_js_1.stringEnum)(ReasonCode, true);
