// @paypal/paypal-server-sdk/dist/cjs/models/refundRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.refundRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
var refundPaymentInstruction_js_1 = require("./refundPaymentInstruction.js");
exports.refundRequestSchema = (0, schema_js_1.object)({
    amount: ['amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    noteToPayer: ['note_to_payer', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paymentInstruction: [
        'payment_instruction',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return refundPaymentInstruction_js_1.refundPaymentInstructionSchema; })),
    ],
});
