// @paypal/paypal-server-sdk/dist/cjs/models/refund.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.refundSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var money_js_1 = require("./money.js");
var payeeBase_js_1 = require("./payeeBase.js");
var refundStatus_js_1 = require("./refundStatus.js");
var refundStatusDetails_js_1 = require("./refundStatusDetails.js");
var sellerPayableBreakdown_js_1 = require("./sellerPayableBreakdown.js");
exports.refundSchema = (0, schema_js_1.object)({
    status: ['status', (0, schema_js_1.optional)(refundStatus_js_1.refundStatusSchema)],
    statusDetails: [
        'status_details',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return refundStatusDetails_js_1.refundStatusDetailsSchema; })),
    ],
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    amount: ['amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    acquirerReferenceNumber: ['acquirer_reference_number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    noteToPayer: ['note_to_payer', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    sellerPayableBreakdown: [
        'seller_payable_breakdown',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return sellerPayableBreakdown_js_1.sellerPayableBreakdownSchema; })),
    ],
    payer: ['payer', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return payeeBase_js_1.payeeBaseSchema; }))],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
