// @paypal/paypal-server-sdk/dist/cjs/models/searchResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.searchResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var transactionDetails_js_1 = require("./transactionDetails.js");
exports.searchResponseSchema = (0, schema_js_1.object)({
    transactionDetails: [
        'transaction_details',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return transactionDetails_js_1.transactionDetailsSchema; }))),
    ],
    accountNumber: ['account_number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    startDate: ['start_date', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    endDate: ['end_date', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    lastRefreshedDatetime: ['last_refreshed_datetime', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    page: ['page', (0, schema_js_1.optional)((0, schema_js_1.number)())],
    totalItems: ['total_items', (0, schema_js_1.optional)((0, schema_js_1.number)())],
    totalPages: ['total_pages', (0, schema_js_1.optional)((0, schema_js_1.number)())],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
