// @paypal/paypal-server-sdk/dist/cjs/models/sellerPayableBreakdown.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.sellerPayableBreakdownSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
var netAmountBreakdownItem_js_1 = require("./netAmountBreakdownItem.js");
var platformFee_js_1 = require("./platformFee.js");
exports.sellerPayableBreakdownSchema = (0, schema_js_1.object)({
    grossAmount: ['gross_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    paypalFee: ['paypal_fee', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    paypalFeeInReceivableCurrency: [
        'paypal_fee_in_receivable_currency',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    netAmount: ['net_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    netAmountInReceivableCurrency: [
        'net_amount_in_receivable_currency',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    platformFees: [
        'platform_fees',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return platformFee_js_1.platformFeeSchema; }))),
    ],
    netAmountBreakdown: [
        'net_amount_breakdown',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return netAmountBreakdownItem_js_1.netAmountBreakdownItemSchema; }))),
    ],
    totalRefundedAmount: [
        'total_refunded_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
});
