// @paypal/paypal-server-sdk/dist/cjs/models/setupTokenRequestPaymentSource.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupTokenRequestPaymentSourceSchema = void 0;
var schema_js_1 = require("../schema.js");
var bankRequest_js_1 = require("./bankRequest.js");
var setupTokenRequestCard_js_1 = require("./setupTokenRequestCard.js");
var vaultApplePayRequest_js_1 = require("./vaultApplePayRequest.js");
var vaultPaypalWalletRequest_js_1 = require("./vaultPaypalWalletRequest.js");
var vaultTokenRequest_js_1 = require("./vaultTokenRequest.js");
var vaultVenmoRequest_js_1 = require("./vaultVenmoRequest.js");
exports.setupTokenRequestPaymentSourceSchema = (0, schema_js_1.object)({
    card: ['card', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return setupTokenRequestCard_js_1.setupTokenRequestCardSchema; }))],
    paypal: ['paypal', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultPaypalWalletRequest_js_1.vaultPaypalWalletRequestSchema; }))],
    venmo: ['venmo', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultVenmoRequest_js_1.vaultVenmoRequestSchema; }))],
    applePay: ['apple_pay', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultApplePayRequest_js_1.vaultApplePayRequestSchema; }))],
    token: ['token', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultTokenRequest_js_1.vaultTokenRequestSchema; }))],
    bank: ['bank', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return bankRequest_js_1.bankRequestSchema; }))],
});
