// @paypal/paypal-server-sdk/dist/cjs/models/setupTokenRequestCard.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupTokenRequestCardSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var cardBrand_js_1 = require("./cardBrand.js");
var vaultCardExperienceContext_js_1 = require("./vaultCardExperienceContext.js");
var vaultCardVerificationMethod_js_1 = require("./vaultCardVerificationMethod.js");
exports.setupTokenRequestCardSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    number: ['number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    expiry: ['expiry', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    securityCode: ['security_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    brand: ['brand', (0, schema_js_1.optional)(cardBrand_js_1.cardBrandSchema)],
    billingAddress: ['billing_address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    verificationMethod: [
        'verification_method',
        (0, schema_js_1.optional)(vaultCardVerificationMethod_js_1.vaultCardVerificationMethodSchema),
    ],
    experienceContext: [
        'experience_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultCardExperienceContext_js_1.vaultCardExperienceContextSchema; })),
    ],
});
