// @paypal/paypal-server-sdk/dist/cjs/models/setupTokenResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.setupTokenResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var customer_js_1 = require("./customer.js");
var linkDescription_js_1 = require("./linkDescription.js");
var paymentTokenStatus_js_1 = require("./paymentTokenStatus.js");
var setupTokenResponsePaymentSource_js_1 = require("./setupTokenResponsePaymentSource.js");
exports.setupTokenResponseSchema = (0, schema_js_1.object)({
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customer: ['customer', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return customer_js_1.customerSchema; }))],
    status: ['status', (0, schema_js_1.optional)(paymentTokenStatus_js_1.paymentTokenStatusSchema)],
    paymentSource: [
        'payment_source',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return setupTokenResponsePaymentSource_js_1.setupTokenResponsePaymentSourceSchema; })),
    ],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
