// @paypal/paypal-server-sdk/dist/cjs/models/shippingWithTrackingDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.shippingWithTrackingDetailsSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var fulfillmentType_js_1 = require("./fulfillmentType.js");
var orderTrackerResponse_js_1 = require("./orderTrackerResponse.js");
var phoneNumberWithOptionalCountryCode_js_1 = require("./phoneNumberWithOptionalCountryCode.js");
var shippingName_js_1 = require("./shippingName.js");
var shippingOption_js_1 = require("./shippingOption.js");
exports.shippingWithTrackingDetailsSchema = (0, schema_js_1.object)({
    trackers: [
        'trackers',
        (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return orderTrackerResponse_js_1.orderTrackerResponseSchema; }))),
    ],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return shippingName_js_1.shippingNameSchema; }))],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    phoneNumber: [
        'phone_number',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumberWithOptionalCountryCode_js_1.phoneNumberWithOptionalCountryCodeSchema; })),
    ],
    type: ['type', (0, schema_js_1.optional)(fulfillmentType_js_1.fulfillmentTypeSchema)],
    options: ['options', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return shippingOption_js_1.shippingOptionSchema; })))],
    address: ['address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
});
