// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionPricingScheme.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionPricingSchemeSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
var pricingTier_js_1 = require("./pricingTier.js");
var subscriptionPricingModel_js_1 = require("./subscriptionPricingModel.js");
exports.subscriptionPricingSchemeSchema = (0, schema_js_1.object)({
    version: ['version', (0, schema_js_1.optional)((0, schema_js_1.number)())],
    fixedPrice: ['fixed_price', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    pricingModel: ['pricing_model', (0, schema_js_1.optional)(subscriptionPricingModel_js_1.subscriptionPricingModelSchema)],
    tiers: ['tiers', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return pricingTier_js_1.pricingTierSchema; })))],
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
