// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionTransactionDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionTransactionDetailsSchema = void 0;
var schema_js_1 = require("../schema.js");
var captureStatus_js_1 = require("./captureStatus.js");
var subscriptionAmountWithBreakdown_js_1 = require("./subscriptionAmountWithBreakdown.js");
var subscriptionPayerName_js_1 = require("./subscriptionPayerName.js");
exports.subscriptionTransactionDetailsSchema = (0, schema_js_1.object)({
    status: ['status', (0, schema_js_1.optional)(captureStatus_js_1.captureStatusSchema)],
    id: ['id', (0, schema_js_1.string)()],
    amountWithBreakdown: [
        'amount_with_breakdown',
        (0, schema_js_1.lazy)(function () { return subscriptionAmountWithBreakdown_js_1.subscriptionAmountWithBreakdownSchema; }),
    ],
    payerName: [
        'payer_name',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriptionPayerName_js_1.subscriptionPayerNameSchema; })),
    ],
    payerEmail: ['payer_email', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    time: ['time', (0, schema_js_1.string)()],
});
