// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionPatchApplicationContext.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionPatchApplicationContextSchema = void 0;
var schema_js_1 = require("../schema.js");
var experienceContextShippingPreference_js_1 = require("./experienceContextShippingPreference.js");
var paymentMethod_js_1 = require("./paymentMethod.js");
exports.subscriptionPatchApplicationContextSchema = (0, schema_js_1.object)({
    brandName: ['brand_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    locale: ['locale', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    shippingPreference: [
        'shipping_preference',
        (0, schema_js_1.optional)(experienceContextShippingPreference_js_1.experienceContextShippingPreferenceSchema),
    ],
    paymentMethod: [
        'payment_method',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return paymentMethod_js_1.paymentMethodSchema; })),
    ],
    returnUrl: ['return_url', (0, schema_js_1.string)()],
    cancelUrl: ['cancel_url', (0, schema_js_1.string)()],
});
