// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionAmountWithBreakdown.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionAmountWithBreakdownSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
exports.subscriptionAmountWithBreakdownSchema = (0, schema_js_1.object)({
    grossAmount: ['gross_amount', (0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })],
    totalItemAmount: ['total_item_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    feeAmount: ['fee_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    shippingAmount: ['shipping_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    taxAmount: ['tax_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    netAmount: ['net_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
});
