// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionBillingCycle.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionBillingCycleSchema = void 0;
var schema_js_1 = require("../schema.js");
var frequency_js_1 = require("./frequency.js");
var subscriptionPricingScheme_js_1 = require("./subscriptionPricingScheme.js");
var tenureType_js_1 = require("./tenureType.js");
exports.subscriptionBillingCycleSchema = (0, schema_js_1.object)({
    pricingScheme: [
        'pricing_scheme',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriptionPricingScheme_js_1.subscriptionPricingSchemeSchema; })),
    ],
    frequency: ['frequency', (0, schema_js_1.lazy)(function () { return frequency_js_1.frequencySchema; })],
    tenureType: ['tenure_type', tenureType_js_1.tenureTypeSchema],
    sequence: ['sequence', (0, schema_js_1.number)()],
    totalCycles: ['total_cycles', (0, schema_js_1.optional)((0, schema_js_1.number)())],
});
