// @paypal/paypal-server-sdk/dist/cjs/models/subscription.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionSchema = void 0;
var schema_js_1 = require("../schema.js");
var linkDescription_js_1 = require("./linkDescription.js");
var money_js_1 = require("./money.js");
var planDetails_js_1 = require("./planDetails.js");
var subscriber_js_1 = require("./subscriber.js");
var subscriptionBillingInformation_js_1 = require("./subscriptionBillingInformation.js");
exports.subscriptionSchema = (0, schema_js_1.object)({
    id: ['id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    planId: ['plan_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    startTime: ['start_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    quantity: ['quantity', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    shippingAmount: ['shipping_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    subscriber: ['subscriber', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriber_js_1.subscriberSchema; }))],
    billingInfo: [
        'billing_info',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriptionBillingInformation_js_1.subscriptionBillingInformationSchema; })),
    ],
    createTime: ['create_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    updateTime: ['update_time', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customId: ['custom_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    planOverridden: ['plan_overridden', (0, schema_js_1.optional)((0, schema_js_1.boolean)())],
    plan: ['plan', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return planDetails_js_1.planDetailsSchema; }))],
    links: ['links', (0, schema_js_1.optional)((0, schema_js_1.array)((0, schema_js_1.lazy)(function () { return linkDescription_js_1.linkDescriptionSchema; })))],
});
