// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionsCardBrand.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionsCardBrandSchema = exports.SubscriptionsCardBrand = void 0;
var schema_js_1 = require("../schema.js");
/**
 * Enum for SubscriptionsCardBrand
 */
var SubscriptionsCardBrand;
(function (SubscriptionsCardBrand) {
    SubscriptionsCardBrand["Visa"] = "VISA";
    SubscriptionsCardBrand["Mastercard"] = "MASTERCARD";
    SubscriptionsCardBrand["Discover"] = "DISCOVER";
    SubscriptionsCardBrand["Amex"] = "AMEX";
    SubscriptionsCardBrand["Solo"] = "SOLO";
    SubscriptionsCardBrand["Jcb"] = "JCB";
    SubscriptionsCardBrand["Star"] = "STAR";
    SubscriptionsCardBrand["Delta"] = "DELTA";
    SubscriptionsCardBrand["Switch"] = "SWITCH";
    SubscriptionsCardBrand["Maestro"] = "MAESTRO";
    SubscriptionsCardBrand["CbNationale"] = "CB_NATIONALE";
    SubscriptionsCardBrand["Configoga"] = "CONFIGOGA";
    SubscriptionsCardBrand["Confidis"] = "CONFIDIS";
    SubscriptionsCardBrand["Electron"] = "ELECTRON";
    SubscriptionsCardBrand["Cetelem"] = "CETELEM";
    SubscriptionsCardBrand["ChinaUnionPay"] = "CHINA_UNION_PAY";
    SubscriptionsCardBrand["Diners"] = "DINERS";
    SubscriptionsCardBrand["Elo"] = "ELO";
    SubscriptionsCardBrand["Hiper"] = "HIPER";
    SubscriptionsCardBrand["Hipercard"] = "HIPERCARD";
    SubscriptionsCardBrand["Rupay"] = "RUPAY";
    SubscriptionsCardBrand["Ge"] = "GE";
    SubscriptionsCardBrand["Synchrony"] = "SYNCHRONY";
    SubscriptionsCardBrand["Eftpos"] = "EFTPOS";
    SubscriptionsCardBrand["Unknown"] = "UNKNOWN";
})(SubscriptionsCardBrand || (exports.SubscriptionsCardBrand = SubscriptionsCardBrand = {}));
/**
 * Schema for SubscriptionsCardBrand
 */
exports.subscriptionsCardBrandSchema = (0, schema_js_1.stringEnum)(SubscriptionsCardBrand, true);
