// @paypal/paypal-server-sdk/dist/cjs/models/subscriptionCardRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.subscriptionCardRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var cardType_js_1 = require("./cardType.js");
var subscriptionsCardAttributes_js_1 = require("./subscriptionsCardAttributes.js");
var subscriptionsCardBrand_js_1 = require("./subscriptionsCardBrand.js");
exports.subscriptionCardRequestSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    number: ['number', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    expiry: ['expiry', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    securityCode: ['security_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    type: ['type', (0, schema_js_1.optional)(cardType_js_1.cardTypeSchema)],
    brand: ['brand', (0, schema_js_1.optional)(subscriptionsCardBrand_js_1.subscriptionsCardBrandSchema)],
    billingAddress: ['billing_address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    attributes: [
        'attributes',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return subscriptionsCardAttributes_js_1.subscriptionsCardAttributesSchema; })),
    ],
});
