// @paypal/paypal-server-sdk/dist/cjs/models/transactionInformation.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.transactionInformationSchema = void 0;
var schema_js_1 = require("../schema.js");
var money_js_1 = require("./money.js");
var paypalReferenceIdType_js_1 = require("./paypalReferenceIdType.js");
exports.transactionInformationSchema = (0, schema_js_1.object)({
    paypalAccountId: ['paypal_account_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    transactionId: ['transaction_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paypalReferenceId: ['paypal_reference_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paypalReferenceIdType: [
        'paypal_reference_id_type',
        (0, schema_js_1.optional)(paypalReferenceIdType_js_1.paypalReferenceIdTypeSchema),
    ],
    transactionEventCode: ['transaction_event_code', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    transactionInitiationDate: [
        'transaction_initiation_date',
        (0, schema_js_1.optional)((0, schema_js_1.string)()),
    ],
    transactionUpdatedDate: ['transaction_updated_date', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    transactionAmount: [
        'transaction_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    feeAmount: ['fee_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    discountAmount: ['discount_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    insuranceAmount: ['insurance_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    salesTaxAmount: ['sales_tax_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    shippingAmount: ['shipping_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    shippingDiscountAmount: [
        'shipping_discount_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    shippingTaxAmount: [
        'shipping_tax_amount',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    otherAmount: ['other_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    tipAmount: ['tip_amount', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    transactionStatus: ['transaction_status', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    transactionSubject: ['transaction_subject', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    transactionNote: ['transaction_note', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paymentTrackingId: ['payment_tracking_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    bankReferenceId: ['bank_reference_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    endingBalance: ['ending_balance', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    availableBalance: ['available_balance', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; }))],
    invoiceId: ['invoice_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    customField: ['custom_field', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    protectionEligibility: ['protection_eligibility', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    creditTerm: ['credit_term', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    creditTransactionalFee: [
        'credit_transactional_fee',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    creditPromotionalFee: [
        'credit_promotional_fee',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return money_js_1.moneySchema; })),
    ],
    annualPercentageRate: ['annual_percentage_rate', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    paymentMethodType: ['payment_method_type', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    instrumentType: ['instrument_type', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    instrumentSubType: ['instrument_sub_type', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
