// @paypal/paypal-server-sdk/dist/cjs/models/vaultVenmoRequest.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.vaultVenmoRequestSchema = void 0;
var schema_js_1 = require("../schema.js");
var paypalPaymentTokenCustomerType_js_1 = require("./paypalPaymentTokenCustomerType.js");
var paypalPaymentTokenUsageType_js_1 = require("./paypalPaymentTokenUsageType.js");
var usagePattern_js_1 = require("./usagePattern.js");
var vaultedDigitalWalletShippingDetails_js_1 = require("./vaultedDigitalWalletShippingDetails.js");
var venmoExperienceContext_js_1 = require("./venmoExperienceContext.js");
exports.vaultVenmoRequestSchema = (0, schema_js_1.object)({
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    usagePattern: ['usage_pattern', (0, schema_js_1.optional)(usagePattern_js_1.usagePatternSchema)],
    shipping: [
        'shipping',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultedDigitalWalletShippingDetails_js_1.vaultedDigitalWalletShippingDetailsSchema; })),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        (0, schema_js_1.optional)((0, schema_js_1.boolean)()),
    ],
    usageType: ['usage_type', (0, schema_js_1.optional)(paypalPaymentTokenUsageType_js_1.paypalPaymentTokenUsageTypeSchema)],
    customerType: [
        'customer_type',
        (0, schema_js_1.optional)(paypalPaymentTokenCustomerType_js_1.paypalPaymentTokenCustomerTypeSchema),
    ],
    experienceContext: [
        'experience_context',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return venmoExperienceContext_js_1.venmoExperienceContextSchema; })),
    ],
});
