// @paypal/paypal-server-sdk/dist/cjs/models/vaultedDigitalWalletShippingDetails.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.vaultedDigitalWalletShippingDetailsSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var fulfillmentType_js_1 = require("./fulfillmentType.js");
var phoneNumberWithCountryCode_js_1 = require("./phoneNumberWithCountryCode.js");
var shippingName_js_1 = require("./shippingName.js");
exports.vaultedDigitalWalletShippingDetailsSchema = (0, schema_js_1.object)({
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return shippingName_js_1.shippingNameSchema; }))],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    phoneNumber: [
        'phone_number',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumberWithCountryCode_js_1.phoneNumberWithCountryCodeSchema; })),
    ],
    type: ['type', (0, schema_js_1.optional)(fulfillmentType_js_1.fulfillmentTypeSchema)],
    address: ['address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
});
