// @paypal/paypal-server-sdk/dist/cjs/models/vaultCardExperienceContext.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.vaultCardExperienceContextSchema = void 0;
var schema_js_1 = require("../schema.js");
var vaultInstructionAction_js_1 = require("./vaultInstructionAction.js");
var vaultUserAction_js_1 = require("./vaultUserAction.js");
exports.vaultCardExperienceContextSchema = (0, schema_js_1.object)({
    brandName: ['brand_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    locale: ['locale', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    returnUrl: ['return_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    cancelUrl: ['cancel_url', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    vaultInstruction: [
        'vault_instruction',
        (0, schema_js_1.optional)(vaultInstructionAction_js_1.vaultInstructionActionSchema),
    ],
    userAction: ['user_action', (0, schema_js_1.optional)(vaultUserAction_js_1.vaultUserActionSchema)],
});
