// @paypal/paypal-server-sdk/dist/cjs/models/venmoPaymentTokenUsagePattern.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.venmoPaymentTokenUsagePatternSchema = exports.VenmoPaymentTokenUsagePattern = void 0;
var schema_js_1 = require("../schema.js");
/**
 * Enum for VenmoPaymentTokenUsagePattern
 */
var VenmoPaymentTokenUsagePattern;
(function (VenmoPaymentTokenUsagePattern) {
    VenmoPaymentTokenUsagePattern["Immediate"] = "IMMEDIATE";
    VenmoPaymentTokenUsagePattern["Deferred"] = "DEFERRED";
    VenmoPaymentTokenUsagePattern["RecurringPrepaid"] = "RECURRING_PREPAID";
    VenmoPaymentTokenUsagePattern["RecurringPostpaid"] = "RECURRING_POSTPAID";
    VenmoPaymentTokenUsagePattern["ThresholdPrepaid"] = "THRESHOLD_PREPAID";
    VenmoPaymentTokenUsagePattern["ThresholdPostpaid"] = "THRESHOLD_POSTPAID";
})(VenmoPaymentTokenUsagePattern || (exports.VenmoPaymentTokenUsagePattern = VenmoPaymentTokenUsagePattern = {}));
/**
 * Schema for VenmoPaymentTokenUsagePattern
 */
exports.venmoPaymentTokenUsagePatternSchema = (0, schema_js_1.stringEnum)(VenmoPaymentTokenUsagePattern, true);
