// @paypal/paypal-server-sdk/dist/cjs/models/venmoWalletResponse.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.venmoWalletResponseSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var name_js_1 = require("./name.js");
var phoneNumber_js_1 = require("./phoneNumber.js");
var returnFlow_js_1 = require("./returnFlow.js");
var venmoWalletAttributesResponse_js_1 = require("./venmoWalletAttributesResponse.js");
exports.venmoWalletResponseSchema = (0, schema_js_1.object)({
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    accountId: ['account_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    userName: ['user_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return name_js_1.nameSchema; }))],
    phoneNumber: ['phone_number', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneNumber_js_1.phoneNumberSchema; }))],
    address: ['address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    returnFlow: ['return_flow', (0, schema_js_1.optional)(returnFlow_js_1.returnFlowSchema)],
    attributes: [
        'attributes',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return venmoWalletAttributesResponse_js_1.venmoWalletAttributesResponseSchema; })),
    ],
});
