// @paypal/paypal-server-sdk/dist/cjs/models/venmoWalletVaultAttributes.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.venmoWalletVaultAttributesSchema = void 0;
var schema_js_1 = require("../schema.js");
var storeInVaultInstruction_js_1 = require("./storeInVaultInstruction.js");
var venmoPaymentTokenCustomerType_js_1 = require("./venmoPaymentTokenCustomerType.js");
var venmoPaymentTokenUsagePattern_js_1 = require("./venmoPaymentTokenUsagePattern.js");
var venmoPaymentTokenUsageType_js_1 = require("./venmoPaymentTokenUsageType.js");
exports.venmoWalletVaultAttributesSchema = (0, schema_js_1.object)({
    storeInVault: ['store_in_vault', storeInVaultInstruction_js_1.storeInVaultInstructionSchema],
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    usagePattern: [
        'usage_pattern',
        (0, schema_js_1.optional)(venmoPaymentTokenUsagePattern_js_1.venmoPaymentTokenUsagePatternSchema),
    ],
    usageType: ['usage_type', venmoPaymentTokenUsageType_js_1.venmoPaymentTokenUsageTypeSchema],
    customerType: [
        'customer_type',
        (0, schema_js_1.optional)(venmoPaymentTokenCustomerType_js_1.venmoPaymentTokenCustomerTypeSchema),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        (0, schema_js_1.optional)((0, schema_js_1.boolean)()),
    ],
});
