// @paypal/paypal-server-sdk/dist/cjs/models/venmoPaymentToken.js
"use strict";
/**
 * PayPal Server SDKLib
 *
 * This file was automatically generated by APIMATIC v3.0 ( https://www.apimatic.io ).
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.venmoPaymentTokenSchema = void 0;
var schema_js_1 = require("../schema.js");
var address_js_1 = require("./address.js");
var name_js_1 = require("./name.js");
var paypalPaymentTokenCustomerType_js_1 = require("./paypalPaymentTokenCustomerType.js");
var paypalPaymentTokenUsageType_js_1 = require("./paypalPaymentTokenUsageType.js");
var phoneWithType_js_1 = require("./phoneWithType.js");
var usagePattern_js_1 = require("./usagePattern.js");
var vaultedDigitalWalletShippingDetails_js_1 = require("./vaultedDigitalWalletShippingDetails.js");
exports.venmoPaymentTokenSchema = (0, schema_js_1.object)({
    description: ['description', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    usagePattern: ['usage_pattern', (0, schema_js_1.optional)(usagePattern_js_1.usagePatternSchema)],
    shipping: [
        'shipping',
        (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return vaultedDigitalWalletShippingDetails_js_1.vaultedDigitalWalletShippingDetailsSchema; })),
    ],
    permitMultiplePaymentTokens: [
        'permit_multiple_payment_tokens',
        (0, schema_js_1.optional)((0, schema_js_1.boolean)()),
    ],
    usageType: ['usage_type', (0, schema_js_1.optional)(paypalPaymentTokenUsageType_js_1.paypalPaymentTokenUsageTypeSchema)],
    customerType: [
        'customer_type',
        (0, schema_js_1.optional)(paypalPaymentTokenCustomerType_js_1.paypalPaymentTokenCustomerTypeSchema),
    ],
    emailAddress: ['email_address', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    payerId: ['payer_id', (0, schema_js_1.optional)((0, schema_js_1.string)())],
    name: ['name', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return name_js_1.nameSchema; }))],
    phone: ['phone', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return phoneWithType_js_1.phoneWithTypeSchema; }))],
    address: ['address', (0, schema_js_1.optional)((0, schema_js_1.lazy)(function () { return address_js_1.addressSchema; }))],
    userName: ['user_name', (0, schema_js_1.optional)((0, schema_js_1.string)())],
});
